src/Facility.LanguageServer/FsdDefinitionUtility.cs

```diff
@@ -84,14 +84,10 @@ private static string GetMemberTypeName(this ServiceTypeInfo type)
 			return null;
 		}
 
-		private static string GetFieldTypeName(this ServiceFieldInfo field)
+		private static string GetFieldTypeName(this ServiceInfo service, ServiceFieldInfo field)
 		{
-			return field.TypeName
-				.Replace("[]", "", StringComparison.Ordinal)
-				.Replace("map<", "", StringComparison.Ordinal)
-				.Replace("nullable<", "", StringComparison.Ordinal)
-				.Replace("result<", "", StringComparison.Ordinal)
-				.Replace(">", "", StringComparison.Ordinal);
+			var type = service.GetFieldType(field);
+			return type?.ValueType?.ToString() ?? type?.ToString() ?? field.TypeName;
```

Comment:
You should have a loop that keeps checking `ValueType` until it is `null`. Something like:
```csharp
while (type?.ValueType is { } valueType)
    type = valueType;
```